packages/babel-register/src/babel-core/track-deps.js

```diff
@@ -0,0 +1,94 @@
+"use strict";
+
+const { addHook } = require("pirates");
+
+const instances = new Map();
+
+let counter = 0;
+
+module.exports = exports = load;
+exports.wrap = wrap;
+
+/**
+ * This function exposes a way to load a module and inject logic to track
+ * what dependencies are loaded from there. This includes files that don't
+ * load right away, so the 'files' list should be treated as live data.
+ * If something inside the module triggers a call to `require()`
+ * at a later time, those items will also be included in 'files'.
+ */
+function load(
+  // The require() function to call when loading 'id'.
+  req: string => mixed,
+  // The name of the file that 'id' will be loaded relative to.
+  filename: string,
+  // The module to load.
+  id: string,
+  // A callback to call when resolving a module. If an object is returned,
+  // those exports will be used and that module and anything it loads will not
+  // be tracked in the 'files' Set.
+  overrideCallback: (
+    // The id of a module being loaded.
+    id: string,
+    // The name of the file that is loading the id.
+    filename: string,
+    // 'true' if the 'filename' is known to be in the same package that
+    // was originally passed to 'load'. 'true' as long as the file was
+    // reached using only relative paths within the original module.
+    inRoot: boolean,
+  ) => { exports: mixed } | void,
+) {
+  const files = new Set();
```

Comment:
The set seems to be write-only?

Reply:
It ends up exported and used in `/node.js` to skip them.